core/src/main/scala/cats/Functor.scala

```diff
@@ -177,6 +177,19 @@ trait Functor[F[_]] extends Invariant[F] { self =>
    */
   def tupleRight[A, B](fa: F[A], b: B): F[(A, B)] = map(fa)(a => (a, b))
 
+  /**
+   * Modifies the `A` value in `F[A]` with the supplied function, if the function is defined for the value.
+   * Example:
+   * {{{
+   * scala> import cats.Functor
+   * scala> import cats.implicits.catsStdInstancesForList
+   *
+   * scala> Functor[List].mapOrKeep(List(1, 2, 3)) { case 2 => 42 }
```

Comment:
I would like to suggest a couple of improvements to the scaladoc:
```suggestion
   * scala> import cats.Functor
   * scala> import cats.syntax.all._
   *
   * scala> List(1, 2, 3).mapOrKeep { case 2 => 42 }
```
Since Cats [v2.2.0](https://github.com/typelevel/cats/releases/tag/v2.2.0) it is mostly discouraged to `import cats.implicits._` in favor of `import cats.syntax.all._`

Also, I believe that using extension-based syntax in examples would be easier to understand for users.

Reply:
right now i just followed the other examples in the file. So, i suggest we keep it like that in this PR for uniformity
then i can submit a separate PR rewriting all docs in the file. wdyt?

Reply:
@satorg 
https://github.com/jozic/cats/commit/254499885f974859acc03fe71262d455fc058f09
that  should be fine as a separate PR once this one is merged, right?